pageserver/src/virtual_file.rs

```diff
@@ -1329,7 +1332,8 @@ impl OwnedAsyncWriter for VirtualFile {
         ctx: &RequestContext,
     ) -> std::io::Result<FullSlice<Buf>> {
         let (buf, res) = VirtualFile::write_all_at(self, buf, offset, ctx).await;
-        res.map(|_| buf)
+        let x = res.map(|_| buf).unwrap();
+        Ok(x)
```

Comment:
Why is unwrap ok?

Reply:
4b358f6a3c60b809aa2ff0a2672aa20a5197368a: properly error handling by checking `res?`